care/facility/models/feature_flag.py

```diff
@@ -0,0 +1,34 @@
+from care.facility.registries.feature_flag import FlagRegistry, FlagType
+from care.utils.models.base import BaseModel
+
+from django.db import models
+
+
+class FacilityFlag(BaseModel):
+    facility = models.ForeignKey(
+        "Facility", on_delete=models.CASCADE, null=False, blank=False
+    )
+    flag = models.CharField(max_length=1024)
```

Comment:
Consider using `models.TextChoices`
Ref: https://docs.djangoproject.com/en/5.1/ref/models/fields/#enumeration-types

Reply:
I considered it but the choice field is dyamic in nature, plugins can create more flags but they wont have migrations associated with it as that will create unwanted migration dependencies, I think desigining the validations ourselves is easier than going through that path.

Reply:
Adding more context : Plugins can be added or removed from care at any point, each plugin might have their own feature flag, we should have some mechanism to keep a registry for all these feature flags and maintain them.